vendor/github.com/gardener/etcd-druid/api/v1alpha1/etcd_types.go

```diff
@@ -237,6 +242,20 @@ type SharedConfig struct {
 	AutoCompactionRetention *string `json:"autoCompactionRetention,omitempty"`
 }
 
+// SchedulingConstraints defines the different scheduling constraints that must be applied to the
+// pod spec in the etcd statefulset.
+// Currently supported constraints are Affinity and TopologySpreadConstraints.
+type SchedulingConstraints struct {
+	// Affinity defines the various affinity and anti-affinity rules for a pod
+	// that are honoured by the kube-scheduler.
+	// +optional
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+	// TopologySpreadConstraints describes how a group of pods ought to spread across topology domains,
+	// that are honoured by the kube-scheduler.
+	// +optional
+	TopologySpreadConstraints []corev1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
```

Comment:
TSC is not currently used in `etcd/etcd.go` and in its current form it would not be very useful as we found out earlier. Do you still want to keep it? You can always add it if and when we use it. Right now it will always be nil/empty.

Reply:
I had introduced this field in Etcd spec in druid to allow us the freedom to select which kind of scheduling constraint we want to use in gardener. Maybe not today, but sometime in the future we might decide to use TSC if the upstream k8s community is able to resolve the issues mentioned in https://github.com/kubernetes/kubernetes/issues/109364. 
So I think it doesn't hurt to have this option, to avoid future API changes/additions. And of course, there might always be somebody in the gardener community who might find this useful 😉.